team-telnyx/webrtc
# Add Call Control and other call ids to call state

The RTC backend is going to send `telnyx_call_control_id`,
`telnyx_session_id` and `telnyx_leg_id` params in the
`telnyx_rtc.invite` message. We then save it inside the Call object so
that it is accessible in upcoming `notification` events.

This should enable customers to cross-reference calls between the Call
Control backend and the WebRTC client.

This PR includes a few fixes to the VertoHandler tests.

## 📝 To Do


## ✋ Manual testing

1. `npm link` in this repository
1. `npm link @telnyx/webrtc` in the repository you will run this code on (e.g. call-center/web-client)
1. Open the network inspector and inspect `rtc.telnyx.com` messages
1. Call your webrtc client
1. Inspect the incoming `telnyx_rtc.invite` message params. Confirm it has `telnyx_session_id` and `telnyx_leg_id`. `telnyx_call_control_id` to be implemented.
1. Confirm you got a notification event with `telnyxSessionId` and `telnyxLegId` as call options.

## 🦊 Browser testing

### Desktop


## 📸 Screenshots

| Description | Screenshot |
| ----------- | ---------- |
| Desktop     |            |
| usage.gif   |            |
